### PR TITLE
Feat/3824  autogenerate pre join space message

### PR DESCRIPTION
**Pull request by asf-algoo**
related to #3824 

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: backend/tracim_backend/lib/core/event.py [hunk 1]
```diff
@@ -311,6 +318,52 @@ def create_event(
         context.pending_events.append(event)
         return event
 
+    def generate_historic_workspaces_messages_for_user(
+        self,
+        context: TracimContext,
+        user_id: int,
+        workspace_ids: List[int],
+        max_event_history: int = -1,
```
Comment (grignards) [inline backend/tracim_backend/lib/core/event.py:326]:
```suggestion
        max_messages_count: int = -1,
```

File: backend/tracim_backend/lib/core/event.py [hunk 2]
```diff
@@ -311,6 +318,52 @@ def create_event(
         context.pending_events.append(event)
         return event
 
+    def generate_historic_workspaces_messages_for_user(
```
Comment (grignards) [inline backend/tracim_backend/lib/core/event.py:321]:
```suggestion
    def create_messages_history_for_user(
```

**Review by grignards — CHANGES_REQUESTED** [review]

File: backend/tracim_backend/lib/core/event.py (hunk 1 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/event.py]: done

File: backend/tracim_backend/lib/core/event.py (hunk 2 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/event.py]: done.

**Review by grignards — APPROVED** [review]

**Comment by grignards** [comment]
Tested on branch.

**Comment by inkhey** [comment]
There was an issue with old sqlite version as in our current CI we do used:
- xenial(16.04) on python 3.7 which in travis included sqlite 3.11.
- bionic(18.04) on python 3.6 which in travis included sqlite 3.22.
- bionic(18.04) on python 3.5 which in travis included sqlite 3.22.

The JSON request to filter on workspace does a `JSON_QUOTE` which is an action added to sqlite 3.14, so it was not working with python 3.7 test. A similar sql query already exist in tracim code but is related to only an API test which is runned on python 3.6 only with sqlite 3.22, so we were not able to see the issue here: https://github.com/tracim/tracim/commit/697260b355d47369c485ba361ccceecc4d60579a .

After moving python 3.7 from xenial to bionic, i do get sqlite 3.22 on all database, so with it, we do not have any issue in CI.
We can clearly say we do support sqlite>3.22, and theorically support sqlite>=3.14 with JSON1 extension.

Because of this issue, i added a script for debugging sqlite version/extensions loaded more easily in both CI and dev environnement.

We need to verify which python sqlite version our docker image used, to be sure to not have bug there too.

**Comment by inkhey** [comment]
After some research, it seems that python provided by Gnu/Linux distribution is build with`libsqlite3-0` library by default, which mean the version of this library is the version used by python (at least by default).
This explain why:
- xenial provided sqlite 3.11 : https://packages.ubuntu.com/xenial/libsqlite3-0
- bionic provided sqlite 3.22 : https://packages.ubuntu.com/bionic/libsqlite3-0
- our docker based on stretch use sqlite 3.16.2: https://packages.debian.org/fr/stretch/libsqlite3-0
 (verified manually too).

**Review by grignards — APPROVED** [review]

File: backend/tracim_backend/lib/core/event.py [hunk 3]
```diff
@@ -311,6 +318,54 @@ def create_event(
         context.pending_events.append(event)
         return event
 
+    def create_messages_history_for_user(
+        self, user_id: int, workspace_ids: List[int], max_messages_count: int = -1,
+    ) -> List[Message]:
+        """
+        Generate up to max_messages_count missing messages to ensure the last max_messages_count event
+        related to given workspaces exist as message for user given.
```
Comment (raphj) [inline backend/tracim_backend/lib/core/event.py:326]:
```suggestion
        Generate, for the given user, messages for the last max_messages_count events that happened in the
        given workspace , if they do not already exist for this user.
```

**Review by raphj — APPROVED** [review]

**Pull request merged**